libraries/esp8266/examples/LowPowerDemo/LowPowerDemo.ino

```diff
@@ -0,0 +1,365 @@
+    Serial.println(F("going into Deep Sleep now..."));
+    delay(10);  // sometimes the \n isn't printed without a short delay
+    ESP.deepSleepInstant(10E6, WAKE_NO_RFCAL); // good night!  D0 fires a reset in 10 seconds...
+    delay(10);
+    Serial.println(F("What... I'm not asleep?!?"));  // it will never get here
+  }
+
+  // 9th test - Deep Sleep Instant for 10 seconds, wake with RF_DISABLED
+  if (resetLoop == 3) {  // fourth reset loop since power on
+    resetLoop = 4;  // advance to the next Deep Sleep test after the reset
+    updateRTC();  // save the current test state in RTC memory
+    Serial.println(F("\n9th test - in NO_RFCAL, Deep Sleep Instant for 10 seconds, wake with RF_DISABLED"));
+    float volts = ESP.getVcc();
+    Serial.printf("The internal VCC reads %1.3f volts\n", volts / 1000 );
+    Serial.println(F("press the button to continue"));
+    waitPushbutton(false, blinkDelay);
+    Serial.println(F("going into Deep Sleep now..."));
+    delay(10);  // sometimes the \n isn't printed without a short delay
+    ESP.deepSleepInstant(10E6, WAKE_RF_DISABLED); // good night!  D0 fires a reset in 10 seconds...
+    delay(10);
+    Serial.println(F("What... I'm not asleep?!?"));  // it will never get here
+  }
+
+  if (resetLoop == 4) {
+    resetLoop = 5;  // start all over
+    updateRTC();  // save the current test state in RTC memory
+    float volts = ESP.getVcc();
+    Serial.printf("The internal VCC reads %1.3f volts\n", volts / 1000 );
+    Serial.println(F("\nTests completed, in RF_DISABLED, press the button to do an ESP.restart()"));
+    waitPushbutton(false, 1000);
+    ESP.restart();
+  }
+}
+
+void waitPushbutton(bool usesDelay, unsigned int delayTime) {  // loop until they press the button
+  // note: 2 different modes, as both of the AUTOMATIC power saving modes need a long delay()
+  if (!usesDelay) {  // quick interception of button press, no delay()
+    blinkLED.reset(delayTime);
+    while (digitalRead(WAKE_UP_PIN)) {  // wait for a button press
+      if (blinkLED) {
+        digitalWrite(LED, !digitalRead(LED));  // toggle the activity LED
+        if (WiFi.localIP())  // don't check OTA if WiFi isn't connected
+          ArduinoOTA.handle();  //see if we need to reflash
+      }
+      yield();
+    }
+  } else {  // long delay() for the 2 AUTOMATIC modes, but it misses quick button presses
+    while (digitalRead(WAKE_UP_PIN)) {  // wait for a button press
+      digitalWrite(LED, !digitalRead(LED));  // toggle the activity LED
+      if (WiFi.localIP())  // don't check OTA if WiFi isn't connected
+        ArduinoOTA.handle();  //see if we need to reflash
+      delay(delayTime);
+    }
+  }
+  delay(50);  // debounce time for the switch, button pressed
+  while (!digitalRead(WAKE_UP_PIN))  // now wait for them to release the button
+    delay(10);
+  delay(50);  // debounce time for the switch, button released
+}
+
+uint32_t calculateCRC32(const uint8_t *data, size_t length) {
```

Comment:
It may take me a week to figure that one out, as it's not exactly clear how to use it for a newbie, nor is it listed at readthedocs.  The only example I could find that used it is equally unclear (WiFiShutdown.ino).  At the moment I'm trying to figure out what this is doing, quite literally symbol-by-symbol.
 nv->crc = crc32(&nv->data, sizeof(nv->data));
From what I've figured out so far, that's called "function call by reference" but giving it a name doesn't magically make me understand it.

Reply:
Done, replaced with the one linked from coredecls.h.  That was less painful than I thought.